src/plugins/integrations/AWSIntegration.ts

```diff
@@ -819,4 +257,756 @@ class AWSIntegration implements Integration {
+            get(request, 'params.QueryExecutionContext.Database', ''));
+        const outputLocation: string = get(request, 'params.ResultConfiguration.OutputLocation', '');
+        const operationType = AWSIntegration.getOperationType(operationName, ClassNames.ATHENA);
+        const functionName = InvocationSupport.getFunctionName();
+        const spanName: string = dbName ? dbName : AWS_SERVICE_REQUEST;
+        const parentSpan = tracer.getActiveSpan();
+
+        const activeSpan = tracer._startSpan(spanName, {
+            childOf: parentSpan,
+            domainName: DomainNames.DB,
+            className: ClassNames.ATHENA,
+            disableActiveStart: true,
+            tags: {
+                [SpanTags.OPERATION_TYPE]: operationType,
+                [SpanTags.SPAN_TYPE]: SpanTypes.AWS_ATHENA,
+                [AwsSDKTags.REQUEST_NAME]: operationName,
+            },
+        });
+
+        let queryExecIds: string[] = [];
+        let namedQueryIds: string[] = [];
+
+        if (has(request, 'params.QueryExecutionIds')) {
+            queryExecIds = request.params.QueryExecutionIds;
+        } else if (has(request, 'params.QueryExecutionId')) {
+            queryExecIds = [request.params.QueryExecutionId];
+        }
+
+        if (has(request, 'params.NamedQueryIds')) {
+            namedQueryIds = request.params.NamedQueryIds;
+        } else if (has(request, 'params.NamedQueryId')) {
+            namedQueryIds = [request.params.NamedQueryId];
+        }
+
+        if (operationType) {
+            activeSpan.setTag(SpanTags.TRIGGER_OPERATION_NAMES, [functionName]);
+            activeSpan.setTag(SpanTags.TOPOLOGY_VERTEX, true);
+            activeSpan.setTag(SpanTags.TRIGGER_DOMAIN_NAME, LAMBDA_APPLICATION_DOMAIN_NAME);
+            activeSpan.setTag(SpanTags.TRIGGER_CLASS_NAME, LAMBDA_APPLICATION_CLASS_NAME);
+
+            if (outputLocation !== '') {
+                activeSpan.setTag(AwsAthenaTags.S3_OUTPUT_LOCATION, outputLocation);
+            }
+            if (dbName !== '') {
+                activeSpan.setTag(DBTags.DB_INSTANCE, dbName);
+            }
+            if (!config.maskAthenaStatement) {
+                const query: string = get(request, 'params.QueryString', '');
+                if (query !== '') {
+                    activeSpan.setTag(DBTags.DB_STATEMENT, request.params.QueryString);
+                }
+            }
+            if (queryExecIds.length > 0) {
+                activeSpan.setTag(AwsAthenaTags.REQUEST_QUERY_EXECUTION_IDS, queryExecIds);
+            }
+            if (namedQueryIds.length > 0) {
+                activeSpan.setTag(AwsAthenaTags.REQUEST_NAMED_QUERY_IDS, namedQueryIds);
+            }
+        }
+
+        return activeSpan;
+    }
+
+    public static createTraceLinks(span: ThundraSpan, request: any, config: any): any[] {
+        const response = request.response;
+
+        if (has(response, 'data.QueryExecutionIds')) {
```

Comment:
I think, we should introduce another phase `onResponse` and the tags should be handled there